src/change.jl

```diff
@@ -0,0 +1,33 @@
+
+"""
+    obs = ChangeObservable(val)
+    obs = ChangeObservable{T}(val)
+
+Like an `Observable`, but only updates watchers when the value changes
+"""
+struct ChangeObservable{T} <: AbstractObservable{T}
+    obs::AbstractObservable{T}
+    change::AbstractObservable{T}
+    function ChangeObservable{T}(x) where {T}
+        obs = Observable{T}(x)
+        change = observe_changes(obs)
+        return new{T}(obs, change)
+    end
+end
+ChangeObservable(val::T) where {T} = ChangeObservable{T}(val)
+
+Base.close(co::ChangeObservable) = empty!(listeners(co.change))
```

Comment:
This is defined in Makie for `Observable` which I think is type piracy? Perhaps the other needs bringing across?

Reply:
Oh yeah, would be a nice occasion